### PR TITLE
[tf] make ext-authz server respect headers

### PR DESCRIPTION
**Pull request by nmittler**
The sample external authz server's gRPC implementation will always allow requests if they come from a service account ending in `/sa/a`, even if the header `x-ext-authz=deny`.

This PR does the following:

1. Make the gRPC server respect the `x-ext-authz` header, if provided.
2. Integrating the ext-authz server into the standard Istio build. Since our tests actually use it, this will allow us to more easily update it in the future.

**Please provide a description of this PR:**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -29,6 +29,7 @@ require (
 	github.com/felixge/fgprof v0.9.2
 	github.com/florianl/go-nflog/v2 v2.0.1
 	github.com/fsnotify/fsnotify v1.5.1
+	github.com/gogo/googleapis v1.4.0
```
Comment (howardjohn) [inline go.mod:32]: can you switch the extauthz package to `google.golang.org/genproto/googleapis/rpc/status`? Using `gogo` is both incorrect in the first place and it introduces more dependencies which is not great

File: samples/extauthz/cmd/extauthz/main.go [hunk 2]
```diff
@@ -73,44 +73,50 @@ type ExtAuthzServer struct {
...
-					Headers: []*corev2.HeaderValueOption{
-						{
-							Header: &corev2.HeaderValue{
-								Key:   resultHeader,
-								Value: resultAllowed,
-							},
+}
+
+func (s *extAuthzServerV2) allow(request *authv2.CheckRequest) *authv2.CheckResponse {
+	s.logRequest(true, request)
```
Comment (yangminzhu) [inline samples/extauthz/cmd/extauthz/main.go:88]: I think you can just pass the `allowStr string` directly, and `resultAllowed`/`resultDenied` is also defined so can be used directly.

File: samples/extauthz/cmd/extauthz/main.go [hunk 3]
```diff
@@ -138,48 +144,74 @@ func (s *extAuthzServerV2) Check(ctx context.Context, request *authv2.CheckReque
...
-					Headers: []*corev3.HeaderValueOption{
-						{
-							Header: &corev3.HeaderValue{
-								Key:   resultHeader,
-								Value: resultAllowed,
-							},
+}
+
+func (s *extAuthzServerV3) allow(request *authv3.CheckRequest) *authv3.CheckResponse {
+	s.logRequest(true, request)
```
Comment (yangminzhu) [inline samples/extauthz/cmd/extauthz/main.go:183]: same here.

**Review by yangminzhu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Review by davidhauck — APPROVED** [review]

**Comment by ericvn** [comment]
This PR has caused a number of post-submit failures like: https://storage.googleapis.com/istio-prow/logs/integ-k8s-117_istio_postsubmit/1517038023810748416/build-log.txt
```

2022-04-21T07:15:11.796382Z	info	build complete	runtime=1.134423ms
Error: plan: unknown targets: [ext-authzdocker.app_sidecar_ubuntu_xenial]
make: *** [tools/istio-docker.mk:37: docker.push] Error 255
```

**Pull request merged**